lib/query.js

```diff
@@ -71,7 +82,10 @@ export const isPolynomialTerm = (node) => {
         return true
     } else if (isPow(node)) {
         const [base, exponent] = node.args
-        return (isIdentifier(base) || isPolynomial(base)) && isPolynomialTerm(exponent)
+        const positiveExp = isNumber(exponent)
+              ? getValue(exponent) >= 1
```

Comment:
For it to be a polynomial we want the number to be a positive integer.  Also, this line should be dedent 2 spaces so that the `?` lines up with the `:`.